tests/graph/test-data/rust/src/seeding/producer/data_type.rs

```diff
@@ -46,76 +53,488 @@ pub enum DataTypeProducerConfigError {
+    pub constraints: ValueConstraintsDistributionConfig,
+}
+
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+pub struct MetadataSection {
+    pub provenance: ConstDistributionConfig<ProvidedOntologyEditionProvenance>,
+}
+
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+pub struct ConfigSection {
+    pub conflict_behavior: ConstDistributionConfig<ConflictBehavior>,
+}
+
+/// Domain policy allowing remote and/or local (web-catalog) sources with optional mixing.
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub struct DomainPolicy {
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub remote: Option<RemoteSourceConfig>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub local: Option<LocalSourceConfig>,
+}
+
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub struct RemoteSourceConfig {
+    pub domain: DomainDistributionConfig,
+    #[serde(with = "hash_codec::serde::time")]
+    pub fetched_at: OffsetDateTime,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub weight: Option<u32>,
+}
+
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub struct LocalSourceConfig {
+    pub index: IndexSamplerConfig,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub web_type_weights: Option<LocalWebTypeWeights>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub weight: Option<u32>,
+}
+
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+#[serde(tag = "sampling", rename_all = "camelCase")]
+pub enum IndexSamplerConfig {
+    Uniform,
+}
+
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub struct LocalWebTypeWeights {
+    pub user: u32,
+    pub org: u32,
+}
+
+#[derive(Debug, derive_more::Display)]
+pub enum DomainBindingError {
+    #[display("No source configured")]
+    NoSource,
+    #[display("Missing web catalog for local source")]
+    MissingCatalog,
+    #[display("Local web catalog is empty")]
+    EmptyCatalog,
+    #[display("Invalid weights")]
+    InvalidWeights,
+    #[display("Remote config creation failed")]
+    Remote,
+    #[display("Local web type weights are not supported by this catalog type")]
+    WebTypeWeightsUnsupported,
 }
 
-impl<'c> DataTypeProducerConfig<'c> {
+impl Error for DomainBindingError {}
+
+pub struct ProducerDeps<'c, U: WebCatalog, O: WebCatalog> {
+    pub user_catalog: Option<&'c U>,
+    pub org_catalog: Option<&'c O>,
+}
+
+pub trait WebCatalog: Sync + Send {
+    fn len(&self) -> usize;
+
+    fn is_empty(&self) -> bool {
+        self.len() == 0
+    }
+
+    fn get_entry(&self, index: usize) -> Option<(Arc<str>, Arc<str>, WebId)>;
```

Comment:
Why is the `WebCatalog` indexed by numerical index instead of `WebId`?

Reply:
I'm still experimenting around with that. For property types I used the `choose` approach (PR to be opened). 
I will make the API consistent later.